lib/TPP/DefaultPipeline.cpp

```diff
@@ -124,9 +142,16 @@ struct DefaultPipeline : public tpp::impl::DefaultPipelineBase<DefaultPipeline>,
       pm.addPass(createGpuPipeline(GpuPipelineOptions{gpuBackend}));
     } else {
       // Apply the default preprocessing pass
-      DefaultTppPassesOptions tppDefaultOptions{linalgToLoops,
-                                                parallelTaskGrid};
-      pm.addPass(createDefaultTppPasses(tppDefaultOptions));
+      if (!tileShapeM.empty() && !tileShapeN.empty()) {
+        DefaultTppPassesOptions tppDefaultOptions{
+            linalgToLoops, parallelTaskGrid, tileShapeM,
+            tileShapeN,    shuffleOrder,     outerParallelLoops};
+        pm.addPass(createDefaultTppPasses(tppDefaultOptions));
+      } else {
+        DefaultTppPassesOptions tppDefaultOptions{linalgToLoops,
+                                                  parallelTaskGrid};
```

Comment:
This triggers `-Wmissing-field-initializers`.
I'd suggest creating `DefaultTppPassesOptions` with defaults first, then modifying different options, and adding `createDefaultTppPasses` only once (no branching paths).
This should make it simpler and improve readibility.

Reply:
Fixed with empty Arrayref initial values because I have empty checks in the DefaultTppPasses file. Does that work?

Reply:
I'd collapse the if else:
```mlir
DefaultTppPassesOptions opts;
opts.loops = loops;
...
if(!tileM.empty())
  opts.tileM = tileM;
...
pm.addPass(createDefaultTppPasses(opts));
```